app/components/FlowNavigation.tsx

```diff
@@ -78,58 +60,52 @@ export default function FlowNavigation(props: FlowNavigationProps) {
   );
 }
 
-function isSubflowVisible(navItems: NavItem[], index: number) {
-  return (
-    index === 0 ||
-    [NavState.Open, NavState.Current, NavState.Done].includes(
-      navItems[index].state,
-    )
-  );
-}
-
 function NavItem({
   destination,
   label,
   state,
-  subflows,
+  subflows = [],
   a11yLabels,
   isChild = false,
 }: Readonly<NavItem & { isChild?: boolean }>) {
-  const visibleSubflows = (subflows ?? []).filter((_, index, navItems) =>
-    isSubflowVisible(navItems, index),
+  const visibleChildItems = subflows.filter((subItem) =>
+    stateIsActive(subItem.state),
   );
-
+  const hasSubflows = visibleChildItems.length > 0;
   const isDisabled = stateIsDisabled(state);
-  const isCurrent = state === NavState.Current;
+  const isCurrent = stateIsCurrent(state);
   const isDone = stateIsDone(state);
-  const hasActiveSubflows =
-    stateIsActive(state) &&
-    visibleSubflows.some((subflow) => stateIsActive(subflow.state));
   const collapse = useCollapse({ defaultExpanded: isCurrent });
-  const stateClassNames = classNames(
-    "p-16 flex gap-x-16 items-center ds-label-02-reg hover:underline",
+
+  const liClassNames = classNames("list-none", {
+    "text-gray-600 curser-not-allowed hover:font-normal pointer-events-none":
+      isDisabled,
+    "border-b-[1px] border-b-white last:border-b-0": !isChild,
+    "border-l-[1px] border-l-blue-800": isCurrent,
+  });
+
+  const itemClassNames = classNames(
+    "w-full	ds-label-02-reg p-16 flex gap-x-16 items-center hover:underline hover:bg-blue-300 active:bg-white	focus-visible:shadow-[inset_0px_0px_0px_4px] focus:shadow-blue-800",
```

Comment:
why they have big whitespace? probably tab?
```suggestion
    "w-full ds-label-02-reg p-16 flex gap-x-16 items-center hover:underline hover:bg-blue-300 active:bg-white focus-visible:shadow-[inset_0px_0px_0px_4px] focus:shadow-blue-800",
```